data/transactions/logic/box.go

```diff
@@ -126,113 +169,119 @@ func opBoxReplace(cx *EvalContext) error {
 	start := cx.stack[prev].Uint
 	name := string(cx.stack[pprev].Bytes)
 
-	err := cx.availableBox(name, boxWrite, 0 /* size is already known */)
+	err := argCheck(cx, name, basics.AddSaturate(start, uint64(len(replacement))))
 	if err != nil {
 		return err
 	}
-	box, ok, err := cx.Ledger.GetBox(cx.appID, name)
+
+	contents, exists, err := cx.availableBox(name, boxWrite, 0 /* size is already known */)
 	if err != nil {
 		return err
 	}
-	if !ok {
+	if !exists {
 		return fmt.Errorf("no such box %#v", name)
 	}
 
-	bytes, err := replaceCarefully([]byte(box), replacement, start)
+	bytes, err := replaceCarefully([]byte(contents), replacement, start)
```

Comment:
nit: there are some unnecessary casts to `[]byte` in this file -- I think that's a noop?